### PR TITLE
More Event API

### PR DESCRIPTION
**Pull request by IronCrystal**
I added the following files:

PlayerEnchantItemEvent.java
EntityGameModeChangeEvent.java
EntityShootProjectileEvent.java
EntityVehicleEvent.java

### CODE REVIEW

File: src/main/java/org/spout/vanilla/event/entity/EntityGameModeChangeEvent.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * including the MIT license.
+ */
+package org.spout.vanilla.event.entity;
+
+import org.spout.api.event.HandlerList;
+import org.spout.api.event.Cancellable;
+import org.spout.api.event.entity.EntityEvent;
+import org.spout.api.entity.Entity;
+
+public class EntityGameModeChangeEvent extends EntityEvent implements Cancellable {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/entity/EntityGameModeChangeEvent.java:34]: GameMode's will most likely be undergoing a re factor shortly, so I'll hold of on this for now.

File: src/main/java/org/spout/vanilla/event/player/PlayerEnchantItemEvent.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	@Override
+	public HandlerList getHandlers() {
+		return handlers;
+	}
+
+	public static HandlerList getHandlerList() {
+		return handlers;
+	}
+	
+	public enum Enchantment {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/player/PlayerEnchantItemEvent.java:122]: We want to be able to obtain the Enchantment object, not an enum.

**Comment by IronCrystal** [comment]
So, what do I do?  Do I do another PR with just the three files (not the game mode) and fix the Enchantment object?

**Comment by windy1** [comment]
Sure, that's probably the best course of action since you accidentally included one of @DDoSqc's commits

**Pull request closed without merging**